### PR TITLE
feat: i18n

### PR DESCRIPTION
**Pull request by mrshmllow**
Please squash to PR title

### CODE REVIEW

File: src/components/Goals.astro [hunk 1]
```diff
@@ -1,53 +1,42 @@
 ---
+import { getLangFromUrl, useTranslations } from '../i18n/utils';
 
+const lang = getLangFromUrl(Astro.url);
+const t = useTranslations(lang);
```
Comment (isabelroses) [inline src/components/Goals.astro]: I would prefer to avoid single letter variables/functions

Comment (mrshmllow) [reply src/components/Goals.astro]: Normally I really do agree but its used throughout the pages and I think its fine to be short as you could argue it actually increases readability?

Comment (TheCodedProf) [reply src/components/Goals.astro]: I still think isabel is more correct here. We could call it something like `translation()`, which I'd argue is more readable than `t()`

Comment (liketechnik) [reply src/components/Goals.astro]: The [astro-i18next](https://github.com/yassinedoghri/astro-i18next) extension or [astro's i18n docs](https://docs.astro.build/en/recipes/i18n/) also use `t`. 

In this case I'd value the consistency more, although I generally agree.

Comment (mrshmllow) [reply src/components/Goals.astro]: > The [astro-i18next](https://github.com/yassinedoghri/astro-i18next) extension or [astro's i18n docs](https://docs.astro.build/en/recipes/i18n/) also use t.

Also this, but its fine.

**Comment by mrshmllow** [comment]
I dont think a language switcher is in scope for this PR

File: public/locales/en/translations.json [hunk 2]
```diff
@@ -0,0 +1,51 @@
+{
+	"root.title": "aux.computer",
```
Comment (isabelroses) [inline public/locales/en/translations.json]: I think we can safely assume this title won't change.

Comment (TheCodedProf) [inline public/locales/en/translations.json]: I'm not 100% certain this should be included in the translation file. It's the main domain of the website (for now), and shouldn't be changed in different language versions.

**Comment by nyawox** [comment]
Just my two cents, whether `"header.Github": "Github",` should be included may also be debatable.
Considering GitHub is consistently referred to as GitHub across all supported languages, as evidenced by their [official documentation](https://docs.github.com/en)

File: public/locales/en/translations.json [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	"goals.independent": "We will fork and maintain Nix, NixPkgs, and NixOS. Additional support for projects such as Nix Darwin and Home Manager will be considered.",
+	"goals.gov.title": "Governance",
+	"goals.gov": "A democratic model of governance with elected positions will be used to manage the project. A steering committee will provide direction with additional committees handling specific logistical tasks. Features or individual projects will be owned by Special Interest Groups. SIGs will collaborate within Working Groups to achieve their goals. Each committee and group will have its goals, requirements, and management process clearly defined.",
+	"goals.stabilization.title": "Stabilization",
+	"goals.stabilization": "User-friendly features such as the Nix v3 CLI will be enabled by default. Nix Flakes are used too heavily to remove or change and will instead be stabilized as a v0 in its current state with any future work being handled by a Special Interest Group.",
+	"goals.infa.title": "Infastructure",
+	"goals.infa": "We will build and maintain the tools and infrastructure necessary to keep the ecosystem healthy, including Continuous Integration and Binary Caching.",
```
Comment (Minion3665) [inline public/locales/en/translations.json]: These should probably be 'goals.infra' and 'Infrastructure'
```suggestion
	"goals.infra.title": "Infrastructure",
	"goals.infra": "We will build and maintain the tools and infrastructure necessary to keep the ecosystem healthy, including Continuous Integration and Binary Caching.",
```

File: public/locales/en/translations.json [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+	"roadmap.organization.3": "The `aux` CLI will be modified to provide more ergonomic management of packages and systems. Additional subcommands such as `aux system switch` and `aux system build` will be added to make onboarding and ongoing maintenance easier.",
+	"roadmap.infa.title": "Infastructure",
+	"roadmap.infa.1": "Now that the project has significantly diverged from upstream, we will need to provide our own Continuous Integration and Binary Cache services. Existing governance structures will be used to manage the adoption of these technologies.",
+	"roadmap.infa.2": "Documentation will be included as a part of the Continuous Integration process and will have Continuous Deployment configured to keep it up to date, making it easier for contributors to create educational materials.",
+	"roadmap.infa.3": "Search infrastructure for packages and options will also be created.",
```
Comment (Minion3665) [inline public/locales/en/translations.json]: Ditto here with `infra`
```suggestion
	"roadmap.infra.title": "Infrastructure",
	"roadmap.infra.1": "Now that the project has significantly diverged from upstream, we will need to provide our own Continuous Integration and Binary Cache services. Existing governance structures will be used to manage the adoption of these technologies.",
	"roadmap.infra.2": "Documentation will be included as a part of the Continuous Integration process and will have Continuous Deployment configured to keep it up to date, making it easier for contributors to create educational materials.",
	"roadmap.infra.3": "Search infrastructure for packages and options will also be created.",
```

File: public/locales/en/translations.json (hunk 2 above)
Comment (Minion3665) [inline public/locales/en/translations.json]:
```suggestion
```
I concur with other reviewers

File: src/components/Roadmap.astro [hunk 5]
```diff
@@ -1,102 +1,65 @@
...
-				The `aux` CLI will be modified to provide more ergonomic management of
-				packages and systems. Additional subcommands such as `aux system switch`
-				and `aux system build` will be added to make onboarding and ongoing
-				maintenance easier.
+				{translation("roadmap.organization.3")}
 			</p>
 		</li>
 		<li>
-			<h3>Infrastructure</h3>
+			<h3>{translation("roadmap.infa.title")}</h3>
```
Comment (Minion3665) [inline src/components/Roadmap.astro]:
```suggestion
			<h3>{translation("roadmap.infra.title")}</h3>
```

File: src/components/Roadmap.astro [hunk 6]
```diff
@@ -1,102 +1,65 @@
...
 			<p>
-				Now that the project has significantly diverged from upstream, we will
-				need to provide our own Continuous Integration and Binary Cache
-				services. Existing governance structures will be used to manage the
-				adoption of these technologies.
+				{translation("roadmap.infa.1")}
```
Comment (Minion3665) [inline src/components/Roadmap.astro]:
```suggestion
				{translation("roadmap.infra.1")}
```

File: src/components/Roadmap.astro [hunk 7]
```diff
@@ -1,102 +1,65 @@
...
 			</p>
 			<p>
-				Documentation will be included as a part of the Continuous Integration
-				process and will have Continuous Deployment configured to keep it up to
-				date, making it easier for contributors to create educational materials.
+				{translation("roadmap.infa.2")}
```
Comment (Minion3665) [inline src/components/Roadmap.astro]:
```suggestion
				{translation("roadmap.infra.2")}
```

File: src/components/Roadmap.astro [hunk 8]
```diff
@@ -1,102 +1,65 @@
...
 			</p>
 			<p>
-				Search infrastructure for packages and options will also be created.
+				{translation("roadmap.infa.3")}
```
Comment (Minion3665) [inline src/components/Roadmap.astro]:
```suggestion
				{translation("roadmap.infra.3")}
```

File: src/components/Goals.astro [hunk 9]
```diff
@@ -1,53 +1,42 @@
...
-					User-friendly features such as the Nix v3 CLI will be enabled by
-					default. Nix Flakes are used too heavily to remove or change and will
-					instead be stabilized as a v0 in its current state with any future work
-					being handled by a Special Interest Group.
+					{translation("goals.stabilization")}
 				</p>
 			</li>
 			<li>
-				<h3>Infrastructure</h3>
+				<h3>{translation("goals.infa.title")}</h3>
```
Comment (Minion3665) [inline src/components/Goals.astro]:
```suggestion
				<h3>{translation("goals.infra.title")}</h3>
```

File: src/components/Goals.astro [hunk 10]
```diff
@@ -1,53 +1,42 @@
...
 				<p>
-					We will build and maintain the tools and infrastructure necessary to
-					keep the ecosystem healthy, including Continuous Integration and Binary
-					Caching.
+					{translation("goals.infa")}
```
Comment (Minion3665) [inline src/components/Goals.astro]:
```suggestion
					{translation("goals.infra")}
```

**Review by Minion3665 — DISMISSED** [review]
Generally pretty happy, thanks very much for working on this. I've made some suggestions, feel free to apply them as a batch using GitHub's suggestion UI if you like, or not if you prefer

File: public/locales/en/translations.json (hunk 3 above)
Comment (mrshmllow) [reply public/locales/en/translations.json]: Im never rekeying anything ever again

**Review by TheCodedProf — DISMISSED** [review]
Looks all good to me!

File: public/locales/en/translations.json [hunk 11]
```diff
@@ -27,10 +27,10 @@
...
 	"roadmap.organization.3": "The `aux` CLI will be modified to provide more ergonomic management of packages and systems. Additional subcommands such as `aux system switch` and `aux system build` will be added to make onboarding and ongoing maintenance easier.",
-	"roadmap.infa.title": "Infastructure",
-	"roadmap.infa.1": "Now that the project has significantly diverged from upstream, we will need to provide our own Continuous Integration and Binary Cache services. Existing governance structures will be used to manage the adoption of these technologies.",
-	"roadmap.infa.2": "Documentation will be included as a part of the Continuous Integration process and will have Continuous Deployment configured to keep it up to date, making it easier for contributors to create educational materials.",
-	"roadmap.infa.3": "Search infrastructure for packages and options will also be created.",
+	"roadmap.infra.title": "Infastructure",
```
Comment (Minion3665) [inline public/locales/en/translations.json]: I notice it still says infa here... it's not a rekey so hopefully this is easy to fix, but maybe we should still do so here
```suggestion
	"roadmap.infra.title": "Infrastructure",
```

Comment (mrshmllow) [reply public/locales/en/translations.json]: lmao

**Comment by mrshmllow** [comment]
My vim spell was misconfigured for JSON lol

**Review by Minion3665 — COMMENTED** [review]
+1, looks good to me but somebody else must approve: Thanks mrshmllow, I'm on mobile now so cannot test this immediately but the code looks nice. When someone is able to test it, I would be happy with merging.

**Review by TheCodedProf — APPROVED** [review]
Runs on dev and building it does not error and generates properly. Looks good to me!

**Review by isabelroses — APPROVED** [review]
Working for me

**Pull request merged**